static/app/views/profiling/landing/profilingScatterChart.tsx

```diff
@@ -142,7 +176,22 @@ function makeScatterChartOptions({location, theme}: {location: Location; theme:
       top: 5,
       selected: getSeriesSelection(location),
     },
-    onClick: _params => {}, // TODO
+    onClick: params => {
+      const dataPoint = data[params.seriesName]?.[params.dataIndex];
+      if (!defined(dataPoint)) {
+        return;
+      }
+      const project = projects.find(proj => proj.id === dataPoint.app_id);
+      if (!defined(project)) {
+        return;
+      }
+      const target = generateFlamegraphRoute({
+        orgSlug: organization.slug,
+        projectSlug: project.slug,
+        profileId: dataPoint.id,
+      });
+      browserHistory.push(target);
```

Comment:
Nit: you can call browserHistory with the result of route fn without the local var, it's imo simple enought

```ts
browserHistory.push(generateFlamegraphRoute({
        orgSlug: organization.slug,
        projectSlug: project.slug,
        profileId: dataPoint.id,
}));
```